frontend/src/components/Exercise/ExerciseRuntimeTab.tsx

```diff
@@ -28,92 +23,139 @@ const ExerciseRuntimeTab = (props: any) => {
+                        <ArrowPathIcon className="w-6 h-6" />
+                    </button>
+                )}
             </div>
-            {suffix !== "" ? (<><br />{suffix}</>) : (<></>)}
-        </pre>
 
-    </>)
+            <h5 className="text-2xl font-bold mt-6">Suffix</h5>
+            <p className="text-gray-600 mb-4">This will be appended after the student's code at runtime.</p>
+
+            <div className="relative rounded-2xl">
+                <Editor
+                    className="h-48 p-2 border rounded-lg bg-white shadow-md mb-4 focus:ring-blue-500 focus:border-blue-500"
+                    value={suffix}
+                    onChange={(value, e) => { setSuffix(value as string) }}
+                    language={course?.language?.toLowerCase()}
+                    options={{
+                        minimap: { enabled: false },
+                        lineNumbers: "on",
+                        readOnly: !isOwner,
+                        roundedSelection: false,
+                        scrollbar: {
+                            verticalScrollbarSize: 8,
+                            horizontalScrollbarSize: 8,
+                            useShadows: true,
+                            vertical: "visible",
+                            horizontal: "visible",
+                        },
+                        lineDecorationsWidth: 4,
+                        glyphMargin: true,
+                        renderLineHighlight: "none",
+                        renderFinalNewline: false,
+                        renderLineHighlightOnlyWhenFocus: false,
+                        renderValidationDecorations: "on",
+                        renderWhitespace: "none",
+                    }}
+                />
+                {isOwner && (
+                    <button
+                        className="absolute top-2 right-2 bg-gray-500 hover:bg-gray-600 text-white font-semibold rounded-md py-2 px-4 transition-colors duration-300"
+                        onClick={handleUpdateExercise}
+                    >
+                        <ArrowPathIcon className="w-6 h-6" />
+                    </button>
+                )}
+            </div>
+            {isNotificationVisible && (
+                <div
+                    className={`flex items-center justify-between px-4 py-2 mb-4 rounded ${notificationType === "error" ? "bg-red-500" : "bg-green-500"
+                        } text-white`}
+                >
+                    <span>{notificationMessage}</span>
+                    <button onClick={handleCloseNotification}>Close</button>
+                </div>
+            )}
+        </div>
+
+        <div className="border rounded-lg bg-white shadow-md p-4 mb-4">
+            <h5 className="text-2xl font-bold mb-2">Code preview</h5>
+            <p className="text-gray-600 mb-4">This is what the tested file will look like.</p>
+            <pre className="border rounded-lg bg-gray-100 p-4">
+                {prefix && (
```

Comment:
If the prefix is empty, React will render an empty string as `""` evaluates to `false` in js.
Maybe use safer and more explicit ternary operator `expr ? foo : bar` ?

See [https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx)

Reply:
Done